### PR TITLE
[9.19] Extend support for protected branches, protected tags, runners and deploy keys

### PR DESCRIPTION
**Pull request by MarioLipinski**
Still needs some cleanup....

### CODE REVIEW

**Comment by m1guelpf** [comment]
Ping me when this is ready :)

File: lib/Gitlab/Api/Projects.php [hunk 1]
```diff
@@ -446,6 +446,54 @@ public function removeHook($project_id, $hook_id)
         return $this->delete($this->getProjectPath($project_id, 'hooks/'.$this->encodePath($hook_id)));
     }
 
+    public function protectedBranches($project_id, array $parameters = [])
+    {
+        $resolver = $this->createOptionsResolver();
```
Comment (glensc) [inline lib/Gitlab/Api/Projects.php:451]: does it make create options resolver with no definitions?

**Comment by GrahamCampbell** [comment]
Thanks for your work on this @MarioLipinski. Are you able to finish off the PR please?

**Comment by GrahamCampbell** [comment]
Closing due to inactivity.

**Pull request closed without merging**